packages/nimbus/src/components/combobox/combobox.stories.tsx

```diff
@@ -0,0 +1,2057 @@
+        // Apple and carrot options should be selected
+        await verifyOptionsSelected(["Apple", "Carrot"], true);
+        // All other options should not be selected
+        await verifyOptionsSelected(
+          ["Banana", "Orange", "Broccoli", "Avocado", "Cucumber"],
+          false
+        );
+        // Close popover for next test
+        await userEvent.keyboard("{Escape}");
+      }
+    );
+    await step(
+      "Multi select selected option can be removed by pressing its' tag's remove button",
+      async () => {
+        // Focus combobox
+        multiSelect.focus();
+        // Remove carrot tag
+        const tagList = await getTagList(multiSelect);
+        const carrotTag = tagList.childNodes[1] as HTMLElement;
+        const removeButton = within(carrotTag).getByRole("button", {
+          name: /remove carrot/i,
+        });
+        await userEvent.click(removeButton);
+        // Verify carrot tag is removed
+        await expect(carrotTag).not.toBeInTheDocument();
+        // Verify apple tag exists
+        await verifyTagsExist(multiSelect, ["Apple"]);
+        // Open popover
+        await userEvent.keyboard("{ArrowDown}");
+        // Verify Apple is selected
+        await verifyOptionsSelected(["Apple"], true);
+        // Verify all other options not selected
+        await verifyOptionsSelected(
+          ["Banana", "Orange", "Carrot", "Broccoli", "Avocado", "Cucumber"],
+          false
+        );
+        // Close popover for next test
+        await userEvent.keyboard("{Escape}");
+      }
+    );
+    await step(
+      "Multi select selected option can be removed by selecting its option",
+      async () => {
+        // Focus combobox
+        multiSelect.focus();
+        const tagList = await getTagList(multiSelect);
+        const appleTag = tagList.childNodes[0] as HTMLElement;
+        // Verify appleTag exists
+        await expect(appleTag).toBeInTheDocument();
+        // Open popover
+        await userEvent.keyboard("{ArrowDown}");
+        // Select Apple option
+        await selectOptionsByName(["Apple"]);
+        // Verify apple tag is removed
+        await expect(appleTag).not.toBeInTheDocument();
+        // Verify all options not selected
+        await verifyOptionsSelected(
+          [
+            "Apple",
+            "Banana",
+            "Orange",
+            "Carrot",
+            "Broccoli",
+            "Avocado",
+            "Cucumber",
+          ],
+          false
+        );
+        // Close popover for next test
+        await userEvent.keyboard("{Escape}");
+      }
+    );
+  },
+};
+
+/**
+ * Complex Options with Descriptions Example
+ * Demonstrates:
+ * - handling item objects that do not have an id or name key
+ * - options with rich content including descriptions for both single and multi-select
+ * - complex options with non-standard keys inside sections
+ */
+export const ComplexOptions: Story = {
```

Comment:
So everyone's aware, I am in the process of adding tests for the rest of the stories.  I am starting the review process without them complete because there is _a lot_ going on in this component, and I want to get everyone's feedback on how it's implemented and what does or does not make sense.

So anything above this point is fair game for 'why didnt you test x', but below here is still a work in progress.